instrumentation/opentelemetry-instrumentation-aiokafka/src/opentelemetry/instrumentation/aiokafka/utils.py

```diff
@@ -0,0 +1,247 @@
+
+
+def _enrich_span(
+    span: Span,
+    bootstrap_servers: Union[str, List[str]],
+    topic: str,
+    partition: Optional[int],
+) -> None:
+    if not span.is_recording():
+        return
+
+    span.set_attribute(SpanAttributes.MESSAGING_SYSTEM, "kafka")
+    span.set_attribute(SpanAttributes.MESSAGING_DESTINATION, topic)
+
+    if partition is not None:
+        span.set_attribute(SpanAttributes.MESSAGING_KAFKA_PARTITION, partition)
+
+    span.set_attribute(
+        SpanAttributes.MESSAGING_URL, json.dumps(bootstrap_servers)
+    )
+
+
+def _get_span_name(operation: str, topic: str):
+    return f"{topic} {operation}"
+
+
+def _wrap_send(
+    tracer: Tracer, async_produce_hook: ProduceHookT
+) -> Callable[..., Awaitable[None]]:
+    async def _traced_send(
+        func: Callable[..., Awaitable[None]],
+        instance: aiokafka.AIOKafkaProducer,
+        args: Tuple[Any],
+        kwargs: Dict[str, Any],
+    ) -> None:
+        headers = _extract_send_headers(args, kwargs)
+        if headers is None:
+            headers = []
+            kwargs["headers"] = headers
+
+        topic = _extract_send_topic(args, kwargs)
+        bootstrap_servers = _extract_bootstrap_servers(instance.client)
+        partition = await _extract_send_partition(instance, args, kwargs)
+        span_name = _get_span_name("send", topic)
+        with tracer.start_as_current_span(
+            span_name, kind=trace.SpanKind.PRODUCER
+        ) as span:
+            _enrich_span(span, bootstrap_servers, topic, partition)
+            propagate.inject(
+                headers,
+                context=trace.set_span_in_context(span),
+                setter=_aiokafka_setter,
+            )
+            try:
+                if callable(async_produce_hook):
+                    await async_produce_hook(span, args, kwargs)
+            except Exception as hook_exception:  # pylint: disable=W0703
+                _LOG.exception(hook_exception)
+
+        return await func(*args, **kwargs)
+
+    return _traced_send
+
+
+async def _create_consumer_span(
+    tracer: Tracer,
+    async_consume_hook: ConsumeHookT,
+    record: ConsumerRecord,
+    extracted_context: Context,
+    bootstrap_servers: Union[str, List[str]],
+    args: Tuple[Any],
+    kwargs: Dict[str, Any],
+):
+    span_name = _get_span_name("receive", record.topic)
+    with tracer.start_as_current_span(
+        span_name,
+        context=extracted_context,
+        kind=trace.SpanKind.CONSUMER,
+    ) as span:
+        new_context = trace.set_span_in_context(span, extracted_context)
+        token = context.attach(new_context)
+        _enrich_span(span, bootstrap_servers, record.topic, record.partition)
+        try:
+            if callable(async_consume_hook):
+                await async_consume_hook(span, record, args, kwargs)
+        except Exception as hook_exception:  # pylint: disable=W0703
+            _LOG.exception(hook_exception)
+        context.detach(token)
```

Comment:
```suggestion
        finally:
            context.detach(token)
```

Reply:
The behaviour would be the same since it's outside the try block no?